packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -0,0 +1,493 @@
+import 'package:flutter/material.dart';
+
+import '../trees.dart';
+
+class Treemap extends StatefulWidget {
+  const Treemap({
+    @required this.rootNode,
+    @required this.levelsVisible,
+    @required this.width,
+    @required this.height,
+    @required this.onTap,
+  });
+
+  final TreemapNode rootNode;
+
+  // The depth of children visible from this treemap widget.
+  // For example, levelsVisible = 2
+  // ---------------
+  // |    l = 1    |
+  // |  ---------  |
+  // |  | l = 2 |  |
+  // |  |       |  |
+  // |  |       |  |
+  // |  ---------  |
+  // ---------------
+  final int levelsVisible;
+
+  final double width;
+  final double height;
+
+  final VoidCallback onTap;
+
+  @override
+  _TreemapState createState() => _TreemapState();
+}
+
+enum PivotType { pivotByMiddle, pivotBySize }
+
+class _TreemapState extends State<Treemap> {
+  PivotType pivotType = PivotType.pivotBySize;
+  static const double minHeightToDisplayText = 20.0;
+  bool shouldDisplayText;
+
+  TreemapNode rootNode;
+
+  @override
+  void initState() {
+    super.initState();
+    rootNode = widget.rootNode;
+    shouldDisplayText = widget.height > minHeightToDisplayText;
+  }
+
+  void cellOnTap(String name) {
+    final TreemapNode child = rootNode.childrenMap[name];
+    if (child != null) {
+      setState(() {
+        rootNode = child;
+      });
+    }
+  }
+
+  // Computes the total size of a given list of treemap nodes.
+  int computeSize(List<TreemapNode> children, int start, int end) {
+    int sum = 0;
+    for (int i = start; i <= end; i++) {
+      sum += children[i].byteSize;
+    }
+    return sum;
+  }
+
+  int computePivot2(List<TreemapNode> children) {
+    switch (pivotType) {
+      case PivotType.pivotByMiddle:
+        {
```

Comment:
you can remove the brackets for each case

Reply:
Done